### DISCUSSION TITLE
desactivate the particles when they leave the hydrodynamic model in the open ocean

### DISCUSSION THREAD

**Discussion by Boorhin** [OpenDrift/opendrift#1158] [Q&A]
Hi,
I use the o.set_config('general:use_auto_landmask', True) for my runs
However I don't know how to deactivate the particles that leave the hydrodynamic model when they are in the open ocean. How can the code know that the particle should be deactivated ? There is no mask for open ocean. The result is that particles continue travelling or hang at the border as there is always a nearest neighbour. I have not seen any description on how to do it in the doc. 
Can we set a mask or can we tell to deactivate when at a certain distance from the nearest model node? Or something similar?
Coastal models will have high density meshes close to the coast and slack ones in the distance. That's the only characteristic I am thinking of.
Cheers,

**Comment by knutfrode** [comment]
Hi,
For some models (e.g. OceanDrift), there is a fallback current of 0 m/s, whenever particles are outside of any reader coverage, or if there should be a missing timestep etc. This is sometimes desirable, but sometimes (as in your case) one would rather deactivate particles if current is missing.
You can do this by disabling the fallback/backup values:
```
o.set_config('environment:fallback:x_sea_water_velocity', None)
o.set_config('environment:fallback:y_sea_water_velocity', None)
```
And similar for other variables.

**Comment by Boorhin** [comment]
I understand what you mean but since the particle will look for a nearest neighbour it will always find a velocity in the environment,  will it not? I don't remember having a maximum distance of interpolation in the readers

**Comment by knutfrode** [comment]
For "structured" readers it will get no data outside of the model coverage area.
However, for unstructured readers I am not quite sure, perhaps @gauteh knows what happens there outside boundary?

**Comment by Boorhin** [comment]
In the unstructured readers I implemented I never saw it but it could be that I missed something.
Thanks for the help. What I know was done was to implement a third value for the landmask but I think that removing the binary type from it is not ideal we would be better with another binary mask or a distance threshold 
or a shapefile maybe?
